zcash_client_backend/src/proto.rs

```diff
@@ -751,18 +757,27 @@ impl proposal::Proposal {
                                             .map_err(ProposalDecodingError::MemoInvalid)
                                     })
                                     .transpose()?;
-                                match cv.pool_type()? {
-                                    PoolType::Shielded(ShieldedProtocol::Sapling) => {
+                                match (cv.pool_type()?, cv.is_ephemeral) {
+                                    (PoolType::Shielded(ShieldedProtocol::Sapling), false) => {
```

Comment:
I'd extract this transformation into a function that can better documented and tested. It seems that modifying this in the future might be quite risky.

Reply:
Perhaps, but it's following the style of the code around it.